Dropdown/index.jsx

```diff
@@ -249,6 +253,19 @@ const onMouseLeave = (component) => () =>
   })
 
 export default compose(
+  uncontrolled({
+    prop: 'focus',
+    defaultProp: 'autoFocus',
+    handlerName: 'onFocus',
```

Comment:
This is maybe a bit late to the party (especially since I reviewed this implementation once before), but it struck me that maybe the explicit handler/reset duo isn't needed, it could be generalized to this (I _think_):

    handlers: {
      onFocus: () => true,
      onBlur: () => false
    }

Then you'd have the names of the handlers as keys of the `handlers` object, and the handlers readily attached to them. Also you don't need the concept of a resetName + handler, it's all just handlers that affect the prop in some way :) This was just a thought, I think it would work for the current use-cases but haven't gone into depth. Wdyt? I'm not necessarily saying that's a better design or anything, just exploring.